browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -229,11 +231,9 @@ void OnBeforeURLRequestAdBlockTP(const ResponseCallback& next_callback,
   scoped_refptr<base::SequencedTaskRunner> task_runner =
       g_brave_browser_process->ad_block_service()->GetTaskRunner();
 
-  DCHECK(ctx->browser_context);
-
   // DoH or standard DNS queries won't be routed through Tor, so we need to
   // skip it.
-  bool should_check_uncloaked = !ctx->browser_context->IsTor();
+  bool should_check_uncloaked = !(!ctx->browser_context || ctx->browser_context->IsTor());
```

Comment:
why the switch from DCHECK here?

Reply:
https://github.com/brave/brave-core/pull/8707#discussion_r628709600

We don't have a good mocking solution for `ctx->browser_context`, so I have to handle it being `null` in unit tests.